src/merkle/store/tests.rs

```diff
@@ -711,10 +733,66 @@ fn get_leaf_depth_works_with_depth_8() {
     assert_eq!(Err(MerkleError::DepthTooBig(9)), store.get_leaf_depth(root, 8, a));
 }
 
+// SUBSET EXTRACTION
+// ================================================================================================
+
+#[test]
+fn mstore_subset() {
+    // add a Merkle tree of depth 3 to the store
+    let mtree = MerkleTree::new(VALUES8.to_vec()).unwrap();
+    let mut store = MerkleStore::default();
+    let empty_store_num_nodes = store.nodes.len();
+    store.extend(mtree.inner_nodes());
+
+    // build 3 subtrees contained within the above Merkle tree; note that subtree2 is a subset
+    // of subtree1
+    let subtree1 = MerkleTree::new(VALUES8[..4].to_vec()).unwrap();
+    let subtree2 = MerkleTree::new(VALUES8[2..4].to_vec()).unwrap();
+    let subtree3 = MerkleTree::new(VALUES8[6..].to_vec()).unwrap();
```

Comment:
It would be good to understand the behaviour for nested trees.  I.e. when the roots of `subtree1` / `subtree2` / `subtree3` are leaves in `mtree` and then we try and extract `mtree`:

```rust
let substore = store.subset(&[mtree.root()]);
```

Does it extract all 4 trees?  Maybe it would be good to extend this test to evaluate this case?

Reply:
I'm already kind of doing this with `subtree1` and `subtree2`. Here, `subtree2` is nested within `subtree1`. But I've updated the test to make an additional check.

Reply:
good point! I hand't noticed.